exchanges/retry/src/retryExchange.ts

```diff
@@ -117,7 +120,10 @@ export const retryExchange = ({
         if (!maxNumberAttemptsExceeded) {
           // Send failed responses to be retried by calling next on the retry$ subject
           // Exclude operations that have been retried more than the specified max
-          nextRetryOperation(res.operation);
+          nextRetryOperation(
+            retryWith ? retryWith(res.error, res.operation) : res.operation
```

Comment:
Do we expect this to not retry if nothing is returned? Replacing retryIf?

Reply:
I did consider this and may replace this. Mainly, I noticed as well that we would want to defer `retryWhen` potentially to the time it's retried which I think is more important, since it may be used to alter request-time context values